com.unity.ml-agents/Runtime/Academy.cs

```diff
@@ -577,6 +578,8 @@ public void EnvironmentStep()
                 {
                     AgentAct?.Invoke();
                 }
+
+                TeamManagerStep?.Invoke();
```

Comment:
Does this need to be called after `AgentAct?.Invoke()`, or can it possible go on the same delegate? If it needs to go after, can we call it something more generic like `PostAgentAct`?

Reply:
This is basically replacing the MaxStepReached logic in agent. We will want all agents to finish acting first and then call MaxStepReached so I think it needs to go after. Also it should only be invoked once instead of n_agent times so not in the same delegate as AgentAct.
Will change the name.